### DISCUSSION TITLE
Command failed: go mod tidy (go 1.21)

### DISCUSSION THREAD

**Discussion by fdebuire** [renovatebot/renovate#26691] [Request Help] [answered]
### How are you running Renovate?

None

### If you're self-hosting Renovate, select which platform you are using.

None

### What is your question?

Hello,

We are encountering an issue since we upgraded some of our apps to go 1.21. Everything was working fine before when we were on go 1.19.

This is how we run Renovate:
```
npx renovate --endpoint "our Azure-Devops endpoint" --token="our token"
```

We have this in our `config.js`:
```
"postUpdateOptions": ["gomodTidy"]
```

Renovate is properly generating PRs but is only modifying the `go.mod` file and not able to modify the `go.sum` file. Here is the error message:
```
Command failed: go mod tidy
go: go.mod file indicates go 1.21, but maximum version supported by tidy is 1.20
```

It probably means that Renovate is not using the go 1.21 version. Is there any way to fix or configure this? Note that for some of our repos we are still in go 1.19 so Renovate needs to be able to handle multiple go versions.

Thanks

**Comment by viceice** [comment, marked as answer]
you need to run renovate with our docker image, then renovate will be able to install the required go version on the fly. otherwise you need to ensure to install the reguired go version blobally before running renovate.

**Reply by danramosf** [reply]
By `our docker image` do you mean the [renovate/go](https://hub.docker.com/r/renovate/go/tags) or the [renovate/renovate](https://hub.docker.com/r/renovate/renovate) image?

**Comment by rarkins** [comment]
If you're running renovate via npx it means that whatever go version you have on the underlying operating system is installed by you, not Renovate. Renovate just calls out to the file system for the default version of go installed (or fails if it's not installed).

**Reply by fdebuire** [reply]
Thanks for the quick answers, I somehow believed that Renovate was taking care of installing go when it needed it. I will try using the Docker image as suggested.
